openid-federation-1_0.xml

```diff
@@ -761,9 +761,179 @@
+	      </t>
+	      <t>
+		If the <spanx style="verb">crit</spanx> Claim is present,
+		then each array element in this claim's value
+		MUST be a string representing an Entity Statement claim
+		that is not defined by this specification
+		and that claim MUST be understood
+		and be able to be processed by the implementation.
+	      </t>
+	      <t>
+		If the <spanx style="verb">metadata</spanx> Claim is present,
+		verify that its value is syntactially correct,
+		not using <spanx style="verb">null</spanx> as metadata values,
+		as specified in <xref target="metadata"/>.
+	      </t>
+	      <t>
+		If the <spanx style="verb">metadata_policy</spanx> Claim is present,
+		the Entity Statement MUST NOT be an Entity Configuration.
+		Verify that its value is syntactially correct,
+		as specified in <xref target="metadata_policy"/>.
+	      </t>
+	      <t>
+		If the <spanx style="verb">metadata_policy_crit</spanx> Claim is present,
+		the Entity Statement MUST NOT be an Entity Configuration.
+		Each array element in this claim's value
+		MUST be a string representing a Metadata Policy operator
+		that is not defined by this specification
+		and that operator MUST be understood
+		and be able to be processed by the implementation.
+	      </t>
+	      <t>
+		If the <spanx style="verb">constraints</spanx> Claim is present,
+		the Entity Statement MUST NOT be an Entity Configuration.
+		Verify that its value is syntactically correct,
+		as specified in <xref target="chain_constraints"/>.
+	      </t>
+	      <t>
+		Obtain the Entity Configuration for the issuing Entity -
+		the Entity with the Issuer Identifer found in the Entity Statement's
+		<spanx style="verb">iss</spanx> (issuer) Claim.
+		When the <spanx style="verb">iss</spanx> and
+		<spanx style="verb">sub</spanx> Claim values match, this is
+		the Entity Statement being validated itself.
+		Otherwise, this can be obtained either from a Trust Chain or
+		by retrieving it as described in
+		<xref target="federation_configuration"/>.
+	      </t>
+	      <t>
+		The Entity Statement's
+		<spanx style="verb">kid</spanx> (Key ID) header parameter value
+		MUST be a non-zero length string and
+		MUST exactly match the <spanx style="verb">kid</spanx> value
+		for a key in the <spanx style="verb">jwks</spanx> (JWK Set) Claim
+		of the Entity Configuration of the issuing Entity.
+	      </t>
+	      <t>
+		The Entity Statement's signature MUST validate using
+		the issuing Entity's key identified by the
+		<spanx style="verb">kid</spanx> value.
+	      </t>
+	      <t>
+		If the <spanx style="verb">trust_marks</spanx> Claim is present,
+		for each element of the array, validate that the
+		<spanx style="verb">trust_mark_type</spanx> member value matches the
+		<spanx style="verb">trust_mark_type</spanx> Claim value in the
+		Trust Mark JWT that is the value of the
+		<spanx style="verb">trust_mark</spanx> member.
+		Validate each Trust Mark as described in
+		<xref target="trust-mark-validation"/>.
```

Comment:
I do not think this should be a requirement for validating entity statements:
I see trust marks as something that a local policy of the validating entity or a federation policy might require on certain entities. i.e. the validator MAY have a list of trust marks it wants to verify. But in my view it does not care about other trust marks that might be present. I do not really see a problem with accepting an entity that publishes an invalid trust mark of type "foobar", if I'm not interested in "foobar" - as long as everything I'm interested in is valid.

I do believe trust mark verification is important, but I believe it is part of establishing trust with an entity not part of verifying an entity statement. Trust mark verification would come after verifying a trust chain.

Reply:
Also according to section 3.2 `trust_marks` can only occur in entity configurations. Other verification steps in this list would verify that the statement is a entity configuration for such claims.

Reply:
I'm fine with requiring validating correct syntax. Requiring to verify that the trust mark types in the object match the type in the JWT requires decoding of the JWT which is some effort. But I could also live with that.

But I'm against requiring to verify all trust marks, this means significant effort. I'm fine with just removing the last sentence `Validate each Trust Mark as described in <xref target="trust-mark-validation"/>.`